packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -105,72 +93,196 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
     function initialize() public initializer {}
 
     function deposit() external payable nonReentrant {
-        uint256 newDeposits = depositsOf[msg.sender] + msg.value;
-        require(msg.sender != TRUSTED_VALIDATOR, "ValidatorPool: trusted validator cannot deposit");
-        require(
-            newDeposits >= MIN_DEPOSIT_AMOUNT,
-            "ValidatorPool: the total deposit is less than minimum amount"
-        );
+        _increaseBalance(msg.sender, msg.value);
+    }
+
+    function withdraw(uint256 _amount) external payable nonReentrant {
+        _decreaseBalance(msg.sender, _amount);
+        (bool success, ) = payable(msg.sender).call{ value: _amount }("");
+        require(success);
+    }
 
-        depositsOf[msg.sender] = newDeposits;
+    /**
+     * @notice Bond asset corresponding to the given output index. This function is called when submitting output.
+     *         Note that the amount to bond should be at least the minimum bonding amount,
+     *         and the output root must also be given for later output root comparison.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     * @param _amount      Amount of bonding.
+     * @param _expiresAt   Expiration time (in seconds) for bonding.
+     */
+    function bond(
+        uint256 _outputIndex,
+        uint128 _amount,
+        uint128 _expiresAt
+    ) external payable {
+        require(msg.sender == address(L2_ORACLE), "ValidatorPool: sender is not L2OutputOracle");
+        require(_amount > 0, "ValidatorPool: amount cannot be 0");
+        require(_expiresAt > block.timestamp, "ValidatorPool: expiration time should be future");
+
+        Types.Bonding storage bonding = bonds[_outputIndex];
+        require(bonding.expiresAt == 0, "ValidatorPool: exists bonding of given output index");
+
+        // Unbond someone's bonding if available.
+        tryUnbond();
+
+        address submitter = getOutputSubmitter(_outputIndex);
+        _decreaseBalance(submitter, _amount);
+
+        bonding.amount = _amount;
+        bonding.expiresAt = _expiresAt;
+
+        emit Bonded(submitter, _outputIndex, _amount, _expiresAt);
+    }
+
+    /**
+     * @notice Increases the amount in a bonding corresponding to the given output index.
+     *         This function is called when creating challenge.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     */
+    function increaseBond(address _challenger, uint256 _outputIndex) external payable {
+        Types.Bonding storage bonding = bonds[_outputIndex];
+        require(bonding.expiresAt > 0, "ValidatorPool: bonding is not exists");
+
+        _decreaseBalance(_challenger, bonding.amount);
+        bonding.amount = bonding.amount * 2;
+
+        emit Increased(_challenger, _outputIndex, bonding.amount);
```

Comment:
I understand event `Increased` represents increased amount, not the total amount.
But here `bonding.amount` is doubled value above.

Reply:
good.